ArduCopter/mode_rtl.cpp

```diff
@@ -533,10 +533,17 @@ void ModeRTL::compute_return_target()
     if ((copter.fence.get_enabled_fences() & AC_FENCE_TYPE_ALT_MAX) != 0) {
         // get return target as alt-above-home so it can be compared to fence's alt
         if (rtl_path.return_target.get_alt_cm(Location::AltFrame::ABOVE_HOME, target_alt)) {
-            float fence_alt = copter.fence.get_safe_alt_max()*100.0f;
-            if (target_alt > fence_alt) {
-                // reduce target alt to the fence alt
-                rtl_path.return_target.alt -= (target_alt - fence_alt);
+            float fence_alt;
+            if (copter.fence.get_safe_alt_max(rtl_path.return_target, Location::AltFrame::ABOVE_HOME, fence_alt)) {
+                // convert to cm
+                fence_alt *= 100;
+
+                if (target_alt > fence_alt) {
+                    // reduce target alt to the fence alt
+                    rtl_path.return_target.alt -= (target_alt - fence_alt);
+                }
+            } else {
+                gcs().send_text(MAV_SEVERITY_CRITICAL, "RTL: no terrain data for fence");
```

Comment:
@rmackay9 should we be raising a terrain failsafe here instead?  If so, we should raise an issue for that.

Reply:
We probably can't raise a terrain failsafe because the standard terrain failsafe is to RTL.  Its possible we shouldn't just quietly do a regular RTL though.  My longer term hope is that we will add Terrain to path planning object avoidance and then RTL can use that.